plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitManager.java

```diff
@@ -44,19 +47,25 @@ public class IcebergSplitManager
     private final IcebergTransactionManager transactionManager;
     private final TypeManager typeManager;
     private final TrinoFileSystemFactory fileSystemFactory;
+    private final ExecutorService executorService;
     private final boolean asyncIcebergSplitProducer;
+    private final IcebergConfig config;
```

Comment:
config is mutable, do not store it on a field

Reply:
Removed the field entirely, it's not needed